### DISCUSSION TITLE
Pruning / Tensorflow / API 3.0 / API 2.0

### DISCUSSION THREAD

**Discussion by salehjg** [intel/neural-compressor#1994] [Q&A] [answered]
Hi there,

I am new to neural-compressor and have three questions:

1. Is pruning not available in API 3.0?
2. Is pruning not available through Tensorflow with API 2.0?
3. For SNIP pruning, are the sparsity masks automatically enforced after single-shot pruning before training starts?

Thank you!

**Comment by xin3he** [comment, marked as answer]
Yes, pruning is not picked to 3x. You can still try pruning with 2x API, and raise issue to us if you met.
